x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorPrivileges.java

```diff
@@ -92,5 +104,12 @@ public void maybeMarkOperatorUser(Authentication authentication, ThreadContext t
         public ElasticsearchSecurityException check(String action, TransportRequest request, ThreadContext threadContext) {
             return null;
         }
+
+        @Override
+        public void maybeInterceptRequest(ThreadContext threadContext, TransportRequest request) {
+            if (request instanceof RestoreSnapshotRequest) {
+                ((RestoreSnapshotRequest) request).skipOperatorOnly(false);
```

Comment:
Similarly, this field is always set even when operator privileges are not enabled to retain the default behaviour. See also above.